### PR TITLE
Converted Stat Boost to an Event from Function (Ready to merge)

### PR DESCRIPTION
**Pull request by KDSKardabox**

### CODE REVIEW

File: src/battle_events/stat_boost.c [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        if ((abilities[ability].on_stat_boost_mod) && (ACTIVE_BANK(i))) {
+            add_callback(CB_ON_STAT_BOOST_MOD, 0, 0, i, (u32)abilities[ability].on_stat_boost_mod);
+        }
+    }
+
+    // execute cbs
+    build_execution_order(CB_ON_STAT_BOOST_MOD);
+    battle_master->executing = true;
+    while (battle_master->executing) {
+        pop_callback(bank, stat_id);
```
Comment (EternalCode) [inline src/battle_events/stat_boost.c]: The return type of this callback was changed to bool. However, the return is ignored.

Comment (KDSKardabox) [reply src/battle_events/stat_boost.c]: This is temporary. It will be updated once I get around implementing Clear Body.

Comment (KDSKardabox) [reply src/battle_events/stat_boost.c]: The change seems to break Speed Boost.

Comment (KDSKardabox) [reply src/battle_events/stat_boost.c]: Sometimes it's causing random crashes. Need to investigate

Comment (KDSKardabox) [reply src/battle_events/stat_boost.c]: Fixed the random crashes.

**Pull request merged**